src/server/scripts/Commands/cs_ahbot.cpp

```diff
@@ -73,102 +78,58 @@ class ahbot_commandscript : public CommandScript
-    static bool HandleAHBotItemsAmountQualityCommand(ChatHandler* handler, char const* args)
+    static bool HandleAHBotItemsAmountQualityCommand(ChatHandler* handler, uint32 amount)
     {
-        char* arg = strtok((char*)args, " ");
-        if (!arg)
-            return false;
-        uint32 qualityVal = atoi(arg);
-
-        sAuctionBot->SetItemsAmountForQuality(Q, qualityVal);
-        handler->PSendSysMessage(LANG_AHBOT_ITEMS_AMOUNT, handler->GetTrinityString(ahbotQualityIds[Q]),
+        sAuctionBot->SetItemsAmountForQuality(Q, amount);
+        handler->PSendSysMessage(LANG_AHBOT_ITEMS_AMOUNT, handler->GetTrinityString(ahbotQualityLangIds.at(Q)),
             sAuctionBotConfig->GetConfigItemQualityAmount(Q));
 
         return true;
     }
 
-    static bool HandleAHBotItemsRatioCommand(ChatHandler* handler, char const* args)
+    static bool HandleAHBotItemsRatioCommand(ChatHandler* handler, uint32 alliance, uint32 horde, uint32 neutral)
     {
-        uint32 rVal[MAX_AUCTION_QUALITY];
-        char* arg = strtok((char*)args, " ");
-        for (int i = 0; i < MAX_AUCTION_QUALITY; ++i)
-        {
-            if (!arg)
-                return false;
-            rVal[i] = atoi(arg);
-            arg = strtok(nullptr, " ");
-        }
+        sAuctionBot->SetItemsRatio(alliance, horde, neutral);
 
-        sAuctionBot->SetItemsRatio(rVal[0], rVal[1], rVal[2]);
-
-        for (int i = 0; i < MAX_AUCTION_HOUSE_TYPE; ++i)
-            handler->PSendSysMessage(LANG_AHBOT_ITEMS_RATIO, AuctionBotConfig::GetHouseTypeName(AuctionHouseType(i)), sAuctionBotConfig->GetConfigItemAmountRatio(AuctionHouseType(i)));
+        for (AuctionHouseType type : EnumUtils::Iterate<AuctionHouseType>())
+            handler->PSendSysMessage(LANG_AHBOT_ITEMS_RATIO, AuctionBotConfig::GetHouseTypeName(type), sAuctionBotConfig->GetConfigItemAmountRatio(type));
         return true;
     }
 
     template<AuctionHouseType H>
-    static bool HandleAHBotItemsRatioHouseCommand(ChatHandler* handler, char const* args)
+    static bool HandleAHBotItemsRatioHouseCommand(ChatHandler* handler, uint32 ratio)
     {
-        char* arg = strtok((char*)args, " ");
-        if (!arg)
-            return false;
-        uint32 ratioVal = atoi(arg);
-
-        sAuctionBot->SetItemsRatioForHouse(H, ratioVal);
+        sAuctionBot->SetItemsRatioForHouse(H, ratio);
         handler->PSendSysMessage(LANG_AHBOT_ITEMS_RATIO, AuctionBotConfig::GetHouseTypeName(H), sAuctionBotConfig->GetConfigItemAmountRatio(H));
         return true;
     }
 
-    static bool HandleAHBotRebuildCommand(ChatHandler* /*handler*/, char const* args)
+    static bool HandleAHBotRebuildCommand(ChatHandler* /*handler*/, Optional<ExactSequence<'a', 'l', 'l'>> all)
     {
-        char* arg = strtok((char*)args, " ");
-
-        bool all = false;
-        if (arg && strcmp(arg, "all") == 0)
-            all = true;
-
-        sAuctionBot->Rebuild(all);
+        sAuctionBot->Rebuild(all.has_value());
```

Comment:
```suggestion
        sAuctionBot->Rebuild(all);
```

Reply:
Nope. That does not implicitly convert.

Reply:
Why not? Does `Rebuild` not take `bool`?

Reply:
It does, but still
`error: cannot convert ‘Optional<Trinity::ChatCommands::ExactSequence<'a', 'l', 'l'> >’ {aka ‘std::optional<Trinity::ChatCommands::ExactSequence<'a', 'l', 'l'> >’} to ‘bool’`

Reply:
oh right, the cast operator is explicit